File: filethis-code/android-java/src/main/java/com/filethis/client/model/Destination.java
package com.filethis.client.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Destination {
  /* Human-readable name of the destination. [RO] */
  @JsonProperty("name")
  private String name = null;
  /* Unique Id for the destination. [RO] */
  @JsonProperty("id")
  private Integer id = null;
  /* Whether the destination is exposed to the public or not. [RO] */
  @JsonProperty("state")
  private String state = null;
  /* Type of destination (FileThis Cloud, external provider, or FileThis desktop). [RO] */
  @JsonProperty("type")
  private String type = null;
  /* The storage provider. [RO] */
  @JsonProperty("provider")
  private String provider = null;
  /* Filename of provider's company logo. [RO] */
  @JsonProperty("logo")
  private String logo = null;
  /* Link to the provider's website. [RO] */
  @JsonProperty("url")
  private String url = null;
  /* URL for FileThis help page that describes the destination. [RO] */
  @JsonProperty("helpUrl")
  private String helpUrl = null;
  /* Application Id for iOS mobile app. [RO] */
  @JsonProperty("iosAppId")
  private String iosAppId = null;
  /* Application Id for Android mobile app. [RO] */
  @JsonProperty("androidAppId")
  private String androidAppId = null;
  /* Default web page for mobile app store. [RO] */
  @JsonProperty("appUrl")
  private String appUrl = null;
  /* An ordinal number that suggests a preferred order for display to users. [RO] */
  @JsonProperty("ordinal")
  private Integer ordinal = null;
  /* URL of destination's company logo. [RO] */
  @JsonProperty("logoUrl")
  private String logoUrl = null;
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

  public String getState() {
    return state;
  }
  public void setState(String state) {
    this.state = state;
  }

  public String getType() {
    return type;
  }
  public void setType(String type) {
    this.type = type;
  }

  public String getProvider() {
    return provider;
  }
  public void setProvider(String provider) {
    this.provider = provider;
  }

  public String getLogo() {
    return logo;
  }
  public void setLogo(String logo) {
    this.logo = logo;
  }

  public String getUrl() {
    return url;
  }
  public void setUrl(String url) {
    this.url = url;
  }

  public String getHelpUrl() {
    return helpUrl;
  }
  public void setHelpUrl(String helpUrl) {
    this.helpUrl = helpUrl;
  }

  public String getIosAppId() {
    return iosAppId;
  }
  public void setIosAppId(String iosAppId) {
    this.iosAppId = iosAppId;
  }

  public String getAndroidAppId() {
    return androidAppId;
  }
  public void setAndroidAppId(String androidAppId) {
    this.androidAppId = androidAppId;
  }

  public String getAppUrl() {
    return appUrl;
  }
  public void setAppUrl(String appUrl) {
    this.appUrl = appUrl;
  }

  public Integer getOrdinal() {
    return ordinal;
  }
  public void setOrdinal(Integer ordinal) {
    this.ordinal = ordinal;
  }

  public String getLogoUrl() {
    return logoUrl;
  }
  public void setLogoUrl(String logoUrl) {
    this.logoUrl = logoUrl;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Destination {\n");
    sb.append("  name: ").append(name).append("\n");
    sb.append("  id: ").append(id).append("\n");
    sb.append("  state: ").append(state).append("\n");
    sb.append("  type: ").append(type).append("\n");
    sb.append("  provider: ").append(provider).append("\n");
    sb.append("  logo: ").append(logo).append("\n");
    sb.append("  url: ").append(url).append("\n");
    sb.append("  helpUrl: ").append(helpUrl).append("\n");
    sb.append("  iosAppId: ").append(iosAppId).append("\n");
    sb.append("  androidAppId: ").append(androidAppId).append("\n");
    sb.append("  appUrl: ").append(appUrl).append("\n");
    sb.append("  ordinal: ").append(ordinal).append("\n");
    sb.append("  logoUrl: ").append(logoUrl).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

File: filethis/target-code/csharp/src/Com/FileThis/Model/Destination.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.FileThis.Model {
  public class Destination {
    /* Human-readable name of the destination. [RO] */
    public string name { get; set; }

    /* Unique Id for the destination. [RO] */
    public int? id { get; set; }

    /* Whether the destination is exposed to the public or not. [RO] */
    public string state { get; set; }

    /* Type of destination (FileThis Cloud, external provider, or FileThis desktop). [RO] */
    public string type { get; set; }

    /* The storage provider. [RO] */
    public string provider { get; set; }

    /* Filename of provider's company logo. [RO] */
    public string logo { get; set; }

    /* Link to the provider's website. [RO] */
    public string url { get; set; }

    /* URL for FileThis help page that describes the destination. [RO] */
    public string helpUrl { get; set; }

    /* Application Id for iOS mobile app. [RO] */
    public string iosAppId { get; set; }

    /* Application Id for Android mobile app. [RO] */
    public string androidAppId { get; set; }

    /* Default web page for mobile app store. [RO] */
    public string appUrl { get; set; }

    /* An ordinal number that suggests a preferred order for display to users. [RO] */
    public int? ordinal { get; set; }

    /* URL of destination's company logo. [RO] */
    public string logoUrl { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Destination {\n");
      sb.Append("  name: ").Append(name).Append("\n");
      sb.Append("  id: ").Append(id).Append("\n");
      sb.Append("  state: ").Append(state).Append("\n");
      sb.Append("  type: ").Append(type).Append("\n");
      sb.Append("  provider: ").Append(provider).Append("\n");
      sb.Append("  logo: ").Append(logo).Append("\n");
      sb.Append("  url: ").Append(url).Append("\n");
      sb.Append("  helpUrl: ").Append(helpUrl).Append("\n");
      sb.Append("  iosAppId: ").Append(iosAppId).Append("\n");
      sb.Append("  androidAppId: ").Append(androidAppId).Append("\n");
      sb.Append("  appUrl: ").Append(appUrl).Append("\n");
      sb.Append("  ordinal: ").Append(ordinal).Append("\n");
      sb.Append("  logoUrl: ").Append(logoUrl).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
